mod/embed/start.php

```diff
@@ -49,52 +55,30 @@ function embed_longtext_menu($hook, $type, $items, $vars) {
 		return;
 	}
 
+	$id = elgg_extract('id', $vars);
+	if ($id === null) {
+		return;
+	}
+
 	$url = 'embed';
 
 	$page_owner = elgg_get_page_owner_entity();
 	if (elgg_instanceof($page_owner, 'group') && $page_owner->isMember()) {
-		$url = 'embed?container_guid=' . $page_owner->getGUID();
-	}
-
-	elgg_require_js('jquery.form');
-	elgg_load_js('elgg.embed');
-
-	$text = elgg_echo('embed:media');
-
-	// if loaded through ajax (like on /activity), pull in JS libs manually
-	// hack for #6422 because we haven't converted everything to amd yet
-	if (elgg_in_context('ajax')) {
-		$externals = $GLOBALS['_ELGG']->externals_map;
-		$embed = elgg_extract('elgg.embed', $externals['js']);
-		$lightbox_js = elgg_extract('lightbox', $externals['js']);
-		$lightbox_css = elgg_extract('lightbox', $externals['css']);
-		
-		$text .= <<<___JS
-<script>
-	require(['jquery.form']);
-	if (typeof $.fancybox === 'undefined') {
-		$.getScript('$lightbox_js->url');
-		$('head').append('<link rel="stylesheet" href="$lightbox_css->url"></link>');
-	}
-	if (typeof elgg.embed === 'undefined') {
-		$.getScript('$embed->url');
-	}
-</script>
-___JS;
+		$url = elgg_http_add_url_query_elements($url, [
+			'container_guid' => $page_owner->guid,
+		]);
 	}
 
 	$items[] = ElggMenuItem::factory(array(
 		'name' => 'embed',
-		'href' => 'javascript:void()',
-		'data-colorbox-opts' => json_encode([
-			'href' => elgg_normalize_url($url),
-		]),
-		'text' => $text,
+		'href' => elgg_normalize_url($url),
```

Comment:
This link leads to a dead end page that doesn't let you do anything, hence the previous markup. We don't want the user landing on that page if the event gets bound too late or there's a JS error.